### PR TITLE
 Fix order and user admin list page 

### PR DESCRIPTION
**Pull request by ahsan-ul-haq**
Order and user admin page broken becacuse of large data.
Added switch to get_queryset to not query database

LEARNER-2806

### CODE REVIEW

File: ecommerce/core/admin.py [hunk 1]
```diff
@@ -25,6 +25,12 @@ class EcommerceUserAdmin(UserAdmin):
     )
     show_full_result_count = False
 
+    def get_queryset(self, request):
+        if not waffle.switch_is_active(ORDER_LIST_VIEW_SWITCH):
+            return
```
Comment (clintonb) [inline ecommerce/core/admin.py]: You should always return a `QuerySet`. See [`none()`](https://docs.djangoproject.com/en/1.11/ref/models/querysets/#none).

File: ecommerce/core/admin.py [hunk 2]
```diff
@@ -25,6 +25,12 @@ class EcommerceUserAdmin(UserAdmin):
     )
     show_full_result_count = False
 
+    def get_queryset(self, request):
+        if not waffle.switch_is_active(USER_LIST_VIEW_SWITCH):
+            return User.objects.none()
```
Comment (clintonb) [inline ecommerce/core/admin.py]: Add a message, informing the user what's going on. Do the same for the one below.

```python
# Translators: "Waffle" is the name of a third-party library. It should not be translated
msg = _('User administration has been disabled due to the load on the database. This functionality can be restored by activating the {switch_name} Waffle switch. Be careful when re-activating this switch!').format(USER_LIST_VIEW_SWITCH)

self.message_user(request, msg, level=messages.WARNING)
```

File: ecommerce/core/admin.py [hunk 3]
```diff
@@ -25,6 +25,12 @@ class EcommerceUserAdmin(UserAdmin):
...
+
+        return super(EcommerceUserAdmin, self).get_queryset(request)
+
     def changelist_view(self, request, extra_context=None):
         if not waffle.switch_is_active(USER_LIST_VIEW_SWITCH):
```
Comment (clintonb) [inline ecommerce/core/admin.py]: This is no longer needed. The template can be removed, too.

**Review by clintonb — CHANGES_REQUESTED** [review]

File: ecommerce/core/admin.py (hunk 3 above)
Comment (schenedx) [reply ecommerce/core/admin.py]: If we are using a waffle flag, the template is still needed, right?

**Review by schenedx — COMMENTED** [review]
LGTM after the @clintonb's suggestion is addressed

**Comment by ahsan-ul-haq** [comment]
@clintonb made suggested changes

**Review by schenedx — APPROVED** [review]
👍

File: ecommerce/extensions/order/tests/test_admin.py [hunk 4]
```diff
@@ -27,4 +27,9 @@ def test_changelist_view_disable_switch(self):
         toggle_switch(ORDER_LIST_VIEW_SWITCH, False)
         response = self.client.get(self.order_page_url)
         self.assertEqual(response.status_code, 200)
-        self.assertContains(response, 'List view is temporarily disabled due to large number of records.')
+        self.assertContains(
```
Comment (clintonb) [inline ecommerce/extensions/order/tests/test_admin.py]: There is a proper way to validate messages: https://stackoverflow.com/a/16143864/592820.

File: ecommerce/extensions/order/tests/test_admin.py [hunk 5]
```diff
@@ -26,5 +26,10 @@ def test_changelist_view_disable_switch(self):
         """ Overridden template will load on the list page, if the switch is disabled. """
         toggle_switch(ORDER_LIST_VIEW_SWITCH, False)
         response = self.client.get(self.order_page_url)
+        msg = 'User administration has been disabled due to the load on the database. ' \
+              'This functionality can be restored by activating the {switch_name} Waffle switch. ' \
+              'Be careful when re-activating this switch!'.format(switch_name=ORDER_LIST_VIEW_SWITCH)
         self.assertEqual(response.status_code, 200)
-        self.assertContains(response, 'List view is temporarily disabled due to large number of records.')
+        message = list(response.context.get('messages'))[0]
+        self.assertEqual(message.tags, "warning")
```
Comment (clintonb) [inline ecommerce/extensions/order/tests/test_admin.py]: `self.assertEqual(message.level, messages.WARNING)`

File: ecommerce/extensions/order/tests/test_admin.py [hunk 6]
```diff
@@ -26,5 +26,10 @@ def test_changelist_view_disable_switch(self):
...
+        self.assertTrue(msg in message.message)
```
Comment (clintonb) [inline ecommerce/extensions/order/tests/test_admin.py]: `self.assertEqual(message.message, msg)`

**Review by clintonb — COMMENTED** [review]
Update translations.

**Review by clintonb — COMMENTED** [review]
Include the the .mo and .po files for both the en and es language codes.

**Comment by ahsan-ul-haq** [comment]
@clintonb made suggested changes

File: ecommerce/extensions/order/admin.py [hunk 7]
```diff
@@ -22,18 +24,19 @@ class OrderAdminExtended(OrderAdmin):
     show_full_result_count = False
 
     def get_queryset(self, request):
+        if not waffle.switch_is_active(ORDER_LIST_VIEW_SWITCH):
+            # Translators: "Waffle" is the name of a third-party library. It should not be translated
+            msg = _('User administration has been disabled due to the load on the database. '
```
Comment (clintonb) [inline ecommerce/extensions/order/admin.py]: s/User/Order

**Review by clintonb — APPROVED** [review]

**Comment by clintonb** [comment]
You forgot to update translations after modifying the string for the order admin. No big deal, but please submit an update at your convenience.

**Pull request merged**